modules/platforms/cpp/ignite/client/detail/cluster_connection.cpp

```diff
@@ -212,4 +212,35 @@ std::shared_ptr<node_connection> cluster_connection::get_random_channel() {
     return std::next(m_connections.begin(), idx)->second;
 }
 
+void cluster_connection::perform_request_handler(protocol::client_operation op, transaction_impl *tx,
+    const std::function<void(protocol::writer &)> &wr, const std::shared_ptr<response_handler> &handler) {
+    if (tx) {
+        auto channel = tx->get_connection();
+        if (!channel)
+            throw ignite_error("Transaction was not started properly");
+
+        auto res = channel->perform_request(op, wr, handler);
+        if (!res)
+            throw ignite_error("Connection associated with the transaction is closed");
+
+        return;
+    }
+
+    while (true) {
+        auto channel = get_random_channel();
+        if (!channel)
+            throw ignite_error("No nodes connected");
+
+        auto res = channel->perform_request(op, wr, handler);
```

Comment:
This sends data to the socket, but does not wait for the result, right? So we only retry connection-related issues in this loop?